docs/features/logforwarding/outputs/splunk-forwarding.adoc

```diff
@@ -43,5 +48,113 @@ spec:
       outputRefs:
         - splunk-receiver
 ----
+1. Token points to the secret containing the Splunk HEC token used for authenticating requests.
+2. The base URL of the Splunk instance.
+3. Optional. The name of the index to send events to. If not specified, the default index defined within Splunk is used.
+4. Optional. Compression configuration, available to are: `none`, `gzip`. Default is `none`.
+5. Optional. The source of events sent to this sink.
+6. Optional. Fields to be added to Splunk index.
 
-NOTE: This will forward logs to the log type of the message.  The default index of the splunk server configuration is used when 'index' is not defined
+=== `source`
+In Splunk, the `source` field typically identifies the origin of a log event.
+To ensure consistency and meaningful categorization, the `source` value can be dynamically derived from the `log_type` and `log_source` fields,
+following Cluster Log Forwarder's conventions.
+
+Below the table with default value depends on log_type and log_source will be used if not set in configuration.
+
+|===
+| |pass:[Infrastructure Journal <br/><em>log_type:infrastructure<br/>log_source:node</em>]|pass:[Infrastructure/Application Container<br/><em>log_type:infrastructure\|application<br/>log_source:container</em>]|pass:[Audit<br/><em>log_type:audit<br/>log_source:auditd\|ovn\|openshiftAPI\|kubeAPI<em/>]|Note
+|`source`|SYSLOG_IDENTIFIER*|ns_name_podName_containerName|.log_source|
+|===
```

Comment:
let's add sourceType and payloadkey here as well

Reply:
@jcantrill I forgot add the `host_key`. Will we expose ithttps://vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#host_key

Reply:
As i understand, we decided to NOT expose `sourceType`, no?

Reply:
>As i understand, we decided to NOT expose sourceType, no?

If we are not exposing 'payloadkey' to populate the event then I see no reason to expose 'sourceType'.  We should always set it to '_json' like the example in the splunk documentation.  Additionally we should ensure the table we provide in this document identifies the value

> @jcantrill I forgot add the host_key. Will we expose it https://vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#host_key

IMO there is no need to expose this in our API.  We should either configure the key to be what we already use or set a field to the value of "hostname".  I would prefer to just configure it to be 'hostname'

Reply:
> Additionally we should ensure the table we provide in this document identifies the value

> IMO there is no need to expose this in our API. We should either configure the key to be what we already use or set a field to the value of "hostname". I would prefer to just configure it to be 'hostname'

https://github.com/openshift/cluster-logging-operator/pull/2979/files#diff-46e6d1852418f35b4db6dde8b446604e9d144d0cb4838d378ed8e2609156bca8R168